ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFTimestamp.java

```diff
@@ -84,12 +84,12 @@ public ObjectInspector initialize(ObjectInspector[] arguments) throws UDFArgumen
           "The function TIMESTAMP takes only primitive types");
     }
 
-    if (ss != null && ss.getConf().getBoolVar(ConfVars.HIVE_STRICT_TIMESTAMP_CONVERSION)) {
+    if (ss != null && ss.getConf().getBoolVar(ConfVars.HIVE_STRICT_CHECKS_TYPE_SAFETY)) {
       PrimitiveCategory category = argumentOI.getPrimitiveCategory();
       PrimitiveGrouping group = PrimitiveObjectInspectorUtils.getPrimitiveGrouping(category);
       if (group == PrimitiveGrouping.NUMERIC_GROUP) {
         throw new UDFArgumentException(
-            "Casting NUMERIC types to TIMESTAMP is prohibited (" + ConfVars.HIVE_STRICT_TIMESTAMP_CONVERSION + ")");
+            "Casting NUMERIC types to TIMESTAMP is prohibited (" + ConfVars.HIVE_STRICT_CHECKS_TYPE_SAFETY + ")");
       }
```

Comment:
Should this be here or rather in `TypeCheckProcFactory.DefaultExprProcessor#validateUDF`?